jobs/templates/installation-pipeline.yaml

```diff
@@ -0,0 +1,150 @@
+---
+
+- job:
+    name: installation-pipeline
+    project-type: pipeline
+    description: "Installs Integreatly by using bastion server as Jenkins lave and executing Ansible installer there."
+    sandbox: false
+    parameters:
+      - string:
+          name: REPOSITORY
+          default: https://github.com/integr8ly/installation.git
+          description: "Repository of the Integreatly installer"
+      - string:
+          name: BRANCH
+          default: 'master'
+          description: "Branch of the installer repository"
+      - string:
+          name: GH_CLIENT_ID
+          description: "GitHub Client ID for OAuth Apps, required for some of the walkthroughs. Can be left empty"
+      - string:
+          name: GH_CLIENT_SECRET
+          description: "GitHub Client Secret for OAuth Apps, required for some of the walkthroughs. Can be left empty"
+      - bool:
+          name: SELF_SIGNED_CERTS
+          default: true
+          description: "Indicates whether cluster uses self signed certificates or not"
+      - string:
+          name: ANSIBLE_USER
+          default: ec2-user
+          description: "User for Ansible to access the master node of target cluster"
+      - string:
+          name: MASTER_URLS
+          description: "Comma separated list of URLs for master nodes of target cluster to be used in Ansible inventory file"
+      - string:
+          name: BASTION_USER
+          default: ec2-user
+          description: "User capable of SSH-ing to bastion server"
+      - string:
+          name: BASTION_URL
+          description: "URL for bastion server the target cluster is hidden behind"
+      - string:
+          name: BASTION_PRIVATE_KEY_ID
+          description: "ID of SSH Credentials (private key) used for SSH-ing to bastion"
+      - string:
+          name: PLAYBOOK
+          default: install
+          description: "Playbook to run. Only tested values are 'install' and 'uninstall'"
+    dsl: |
+      import hudson.model.*
+      import jenkins.model.*
+      import hudson.slaves.*
+      import hudson.slaves.EnvironmentVariablesNodeProperty.Entry
+
+      import hudson.plugins.sshslaves.verifiers.*
+
+      String bastionLabel = "${BASTION_URL}-slave"
+
+      timeout(60) {
+          node('cirhos_rhel7') {        
```

Comment:
Shouldn't we parametrize this as well? WDYT?

Reply:
Well we can have differrent machines and labels than only `cirhos_rhel7` or not?

Reply:
Yes, but such slaves will likely serve for specific purposes only. For instance we have various bastion slaves but such slaves you want only use for the pipeline that deals with specific cluster behind that specific bastion. Using 'cirhos_rhel7' is nice limitation to any non-specific slave.

Can you think of any specific use case when one would want to run this pipeline on specific slave?

Reply:
Well I think I can't...I just hope nobody will change our jenkins labels or something like that.:-)